code/common/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required(VERSION 3.20)
+
+set(CMAKE_CXX_STANDARD 23)
+
+project(OpenCloudCommon)
+
+set(
+    SOURCES
+    "clock.cc"
+    "console_logger.cc"
+    "console.cc"
+    "data_stream.cc"
+    "debug.cc"
+    "dynamic_library.cc"
+    "file_helpers.cc"
+    "log.cc"
+    "strings.cc"
+    "synchro.cc"
+)
+
+set(
+    HEADERS
+    "aligned_allocator.h"
+    "bits.h"
+    "clock.h"
+    "console_logger.h"
+    "console.h"
+    "constants.h"
+    "data_stream.h"
+    "debug.h"
+    "dynamic_library.h"
+    "file_helpers.h"
+    "helpers.h"
+    "log.h"
+    "macros.h"
+    "math.h"
+    "platform.h"
+    "scoped_function.h"
+    "strings.h"
+    "synchro.h"
+    "types.h"
+    "window_handle.h"
+)
+
+add_library(${PROJECT_NAME} STATIC ${SOURCES} ${HEADERS})
+target_include_directories (${PROJECT_NAME} PUBLIC ..) # So that the folder name can be used in imports
```

Comment:
This will work but it'll be a bit cleaner to do use an `INTERFACE` library.

```cmake
add_library(ProjectConfiguration INTERFACE)

# global include directories
target_include_directories(ProjectConfiguration INTERFACE 
  "${PROJECT_SOURCE_DIR}/code"
)
```

You can also use this to set the C++ standard globally:
```cmake
# global features
target_compile_features(ProjectConfiguration INTERFACE
  cxx_std_20
)
```

Then you can just link that like any other library and that will be a little more DRY than copying these all over the different projects:
```cmake
target_link_libraries(common PRIVATE
  ProjectConfiguration
)
```

Reply:
Addressed in 0a7fe7869b1ef0b7f90944e172f9b8309e100c48